src/audio/host-legacy.c

```diff
@@ -194,6 +194,8 @@ static uint32_t host_get_copy_bytes_one_shot(struct host_data *hd, struct comp_d
  */
```

Comment:
Please check existing git commits in this project for examples on how to write git commit. SOF follows similar practises as Linux kernel and Zephyr. One git commit per logical change, proper summary line, explains what bug you are fixing, etc

Reply:
please explain what order is incorrect. AFAICT you're only moving the dma_buffer_copy_to() for the one shot case and this will break normal copy.

Reply:
@ranj063 
sorry, I am new to GitHub.

As mentioned in https://github.com/orgs/thesofproject/discussions/9362,

For capture stream scenario, I find a bug that copying order incorrect in Host component.

The copy order should be: Local-buffer => DMA-buffer => Host-buffer.
But actually it is first DMA-buffer => Host-buffer, then Local-buffer => DMA-buffer.

I think host_copy_one_shot() is correct for playback, but incorrect for capture.

So my patch is let host copy change into Local-buffer => DMA-buffer => Host-buffer:
Just move (copy from Local-buffer to DMA-buffer) 
from host_common_update() to host_copy_one_shot().

Reply:
@kv2019i 
sorry, I am new to GitHub.

I think we discuss bug and solution first.

Finally, if the patch is fine I will modify the commit log more normally.

Reply:
@chihualee No problem, that is perfectly ok! In this type of cases, it's usually contion to mark the pull-request with "RFC" (request for comment) tag so signal you are seeking for feedback but the pull request should not be yet ready to be reviewed for merging.